gdx/src/com/badlogic/gdx/utils/JsonSkimmer.java

```diff
@@ -0,0 +1,592 @@
+
+						if (_json_trans_actions[_trans] != 0) {
+							_acts = _json_trans_actions[_trans];
+							_nacts = (int)_json_actions[_acts++];
+							while (_nacts-- > 0) {
+								switch (_json_actions[_acts++]) {
+								case 0:
+								// line 104 "JsonSkimmer.rl"
+								{
+									stringIsName = true;
+								}
+									break;
+								case 1:
+								// line 107 "JsonSkimmer.rl"
+								{
+									String value = new String(data, s, p - s);
+									if (needsUnescape) value = unescape(value);
+									if (stringIsName) {
+										stringIsName = false;
+										if (debug) System.out.println("name: " + value);
+										name = value;
+									} else {
+										if (debug) System.out.println("value: " + name + "=" + value);
+										value(name, value, stringIsUnquoted);
+										name = null;
+									}
+									if (stop) return;
+									stringIsUnquoted = false;
+									s = p;
+								}
+									break;
+								case 2:
+								// line 123 "JsonSkimmer.rl"
+								{
+									if (debug) System.out.println("startObject: " + name);
+									startObject(name);
+									if (stop) return;
+									name = null;
+									{
+										if (top == stack.length) stack = Arrays.copyOf(stack, stack.length * 2);
+										{
+											stack[top++] = cs;
+											cs = 5;
+											_goto_targ = 2;
+											if (true) continue _goto;
+										}
+									}
+								}
+									break;
+								case 3:
+								// line 130 "JsonSkimmer.rl"
+								{
+									if (debug) System.out.println("endObject");
+									pop();
+									if (stop) return;
+									{
+										cs = stack[--top];
+										_goto_targ = 2;
+										if (true) continue _goto;
+									}
+								}
+									break;
+								case 4:
+								// line 136 "JsonSkimmer.rl"
+								{
+									if (debug) System.out.println("startArray: " + name);
+									startArray(name);
+									if (stop) return;
+									name = null;
+									{
+										if (top == stack.length) stack = Arrays.copyOf(stack, stack.length * 2);
+										{
+											stack[top++] = cs;
+											cs = 23;
+											_goto_targ = 2;
+											if (true) continue _goto;
+										}
+									}
+								}
+									break;
+								case 5:
+								// line 143 "JsonSkimmer.rl"
+								{
+									if (debug) System.out.println("endArray");
+									pop();
+									if (stop) return;
+									{
+										cs = stack[--top];
+										_goto_targ = 2;
+										if (true) continue _goto;
+									}
+								}
+									break;
+								case 6:
+								// line 149 "JsonSkimmer.rl"
+								{
+									int start = p - 1;
+									if (data[p++] == '/') {
+										while (p != eof && data[p] != '\n')
+											p++;
+										p--;
+									} else {
+										while (p + 1 < eof && data[p] != '*' || data[p + 1] != '/')
+											p++;
+										p++;
+									}
+									if (debug) System.out.println("comment " + new String(data, start, p - start));
```

Comment:
Should we be using `Gdx.app.debug()` here, instead of `System.out.println()` ?

Reply:
Probably! There's no `print` (without newline) version, which makes the `indent` method more annoying to write. There are many tests that uses sysout, so I'm inclined to leave it. It could be a separate initiative to change all those.

Reply:
Hm, it isn't in a test, but debug probably only gets turned on in tests, so I'm not too worried. It doesn't seem likely that people will be debugging this on platforms that have trouble with System.out .